examples/cluster/plot_cluster_comparison.py

```diff
@@ -147,39 +163,55 @@
     # make connectivity symmetric
     connectivity = 0.5 * (connectivity + connectivity.T)
 
-    # ============
+    # %%
     # Create cluster objects
-    # ============
-    ms = cluster.MeanShift(bandwidth=bandwidth, bin_seeding=True)
+    # ---------------------------------
+
+    # set up clustering models:
+    # 1. Mini Batch KMeans
+    # 2. Affinity Propagation
+    # 3. Mean Shift
+    # 4. Spectral Clustering
+    # 5. Ward
+    # 6. Agglomerative Clustering
+    # 7. DBSCAN
+    # 8. OPTICS
+    # 9. BIRCH
+    # 10. Gaussian Mixture
     two_means = cluster.MiniBatchKMeans(n_clusters=params["n_clusters"], n_init="auto")
-    ward = cluster.AgglomerativeClustering(
-        n_clusters=params["n_clusters"], linkage="ward", connectivity=connectivity
+    affinity_propagation = cluster.AffinityPropagation(
+        damping=params["damping"], preference=params["preference"], random_state=0
     )
+    ms = cluster.MeanShift(bandwidth=bandwidth, bin_seeding=True)
     spectral = cluster.SpectralClustering(
         n_clusters=params["n_clusters"],
         eigen_solver="arpack",
         affinity="nearest_neighbors",
     )
-    dbscan = cluster.DBSCAN(eps=params["eps"])
-    optics = cluster.OPTICS(
-        min_samples=params["min_samples"],
-        xi=params["xi"],
-        min_cluster_size=params["min_cluster_size"],
-    )
-    affinity_propagation = cluster.AffinityPropagation(
-        damping=params["damping"], preference=params["preference"], random_state=0
+    ward = cluster.AgglomerativeClustering(
+        n_clusters=params["n_clusters"], linkage="ward", connectivity=connectivity
     )
     average_linkage = cluster.AgglomerativeClustering(
         linkage="average",
         metric="cityblock",
         n_clusters=params["n_clusters"],
         connectivity=connectivity,
     )
+    dbscan = cluster.DBSCAN(eps=params["eps"])
+    optics = cluster.OPTICS(
+        min_samples=params["min_samples"],
+        xi=params["xi"],
+        min_cluster_size=params["min_cluster_size"],
+    )
     birch = cluster.Birch(n_clusters=params["n_clusters"])
     gmm = mixture.GaussianMixture(
         n_components=params["n_clusters"], covariance_type="full"
     )
 
+    # %%
+    # Define clustering algorithms
+    # ---------------------------------
```

Comment:
This would create a section in the documentation. Then the rendering will not be as expected because you are triggering the section within the `for` loop.

I am not against dissociated the estimator fit and the plotting section. One way would be to create a cell where we create the cluster instance and fit them. Then we need to store them: we can use a dictionary to associate them with the dataset. At each dataset, we can use a new instance using `sklearn.clone` to make sure that we have different instances for the different dataset.

Once all clusterers fitted, we can create a new section that will only make the plotting.